apps/settings/js/panels/screen_lock_passcode/panel.js

```diff
@@ -196,6 +196,9 @@ define(function(require) {
                   this._passcodeBuffer = '';
                   this._updatePassCodeUI();
                   this._showDialogInMode('new');
+                } else if(result === false) {
```

Comment:
`result === false` is equivalent to origin else case. Could you check again if it works by put `updatePassCodeUI` in the `else` case?